app/services/adviser/subject_match_check.rb

```diff
@@ -0,0 +1,51 @@
+class Adviser::SubjectMatchCheck
+  def self.quickfire_subject_match?(degree)
+    new(degree).call
+  end
+
+  def initialize(degree)
+    @degree = degree
+  end
+
+  def call
+    try_subject_uuid || try_exact_name || try_synonyms || try_match_against_input
+  end
+
+private
+
+  def try_subject_uuid
+    @degree.degree_subject_uuid.in? subject_uuids
+  end
+
+  def try_exact_name
+    relevant_subjects.any? do |relevant_subject|
+      (relevant_subject.name.downcase == subject_name)
+    end
+  end
+
+  def try_synonyms
+    relevant_subjects.any? do |relevant_subject|
+      subject_name.in?(relevant_subject.match_synonyms.map(&:downcase)) || # Exact match on any of the synonyms
+        subject_name.in?(relevant_subject.suggestion_synonyms.map(&:downcase)) # || # Exact match on any of the suggestions
```

Comment:
```suggestion
        subject_name.in?(relevant_subject.suggestion_synonyms.map(&:downcase)) # Exact match on any of the suggestions
```
Assuming is a forgotton comment